nengo/utils/progress.py

```diff
@@ -301,6 +514,9 @@ class ProgressUpdater(object):
     """
 
     def __init__(self, progress_bar):
+        warnings.warn(DeprecationWarning(
+            "ProgressUpdater has been deprecated because progress bars "
+            "have been switched to asynchronous updating."))
```

Comment:
I wouldn't consider `ProgressUpdater` to be part of a public API; can we remove this instead of deprecating?

Reply:
[It is referenced in our documentation](https://www.nengo.ai/nengo/nengorc.html), but I'm ok with removing it. In any case, that part of the documentation (and the rc file template) should be updated.